### PR TITLE
fix: resolve N+1 queries and inefficient task scan

### PR DESCRIPTION
**Pull request by sh1nj1**
## Summary
- **N+1 in MessageBuilder**: Replaced per-iteration `Creative.find_by` calls with batch `Creative.where(id:).index_by(&:id)` lookups in `append_context_creatives`, `append_referenced_creative_contexts`, and `collect_rendered_creative_ids`
- **Inefficient task cancellation in Comment**: Scoped `cancel_pending_tasks` query from scanning ALL pending/running/queued tasks globally to only tasks matching `creative_id` (with fallback for legacy tasks where `creative_id IS NULL`)

## Test plan
- [x] All 1655 tests pass (0 failures, 0 errors)
- [x] Rubocop: 0 offenses on changed files

### CODE REVIEW

File: engines/collavre/app/models/collavre/comment.rb [hunk 1]
```diff
@@ -108,8 +108,10 @@ def nullify_selected_version
     end
 
     def cancel_pending_tasks
-      # Cancel tasks triggered by this comment
-      Task.where(status: %w[pending running queued]).each do |task|
+      # Cancel tasks triggered by this comment, scoped to this creative
+      Task.where(status: %w[pending running queued])
+          .where("creative_id = ? OR creative_id IS NULL", creative_id)
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/models/collavre/comment.rb]: ✅ **Valid concern — fixed in 2fd5eb44.**

CommentMoveService#perform_move updates comment.creative_id without touching existing Task records. The creative_id scoping would miss tasks whose creative_id still pointed to the original creative.

**Fix:** Removed creative_id scoping entirely while keeping find_each optimization. The query now matches all pending/running/queued tasks and filters by trigger_event_payload.comment.id — which never changes, regardless of comment moves.

Tests: 1655 runs, 0 failures.

**Comment by sh1nj1** [comment]
@codex review

File: engines/collavre/app/services/collavre/ai_agent/message_builder.rb [hunk 2]
```diff
@@ -109,11 +112,12 @@ def append_referenced_creative_contexts(messages)
         max_depth = 1 + children_level
 
         # Extract creative IDs from markdown links like [title](/creatives/123)
-        content.scan(%r{\[[^\]]*\]\(/creatives/(\d+)\)}).flatten.uniq.each do |id_str|
-          creative_id = id_str.to_i
-          next if @injected_creative_ids.include?(creative_id)
+        referenced_ids = content.scan(%r{\[[^\]]*\]\(/creatives/(\d+)\)}).flatten.uniq.map(&:to_i)
```
[the remark this answers is not in the document]
Comment (sh1nj1) [reply engines/collavre/app/services/collavre/ai_agent/message_builder.rb]: ✅ **Valid concern — fixed in 4363b39c.**

Moved `.uniq` after `.map(&:to_i)` so deduplication happens on integer values, not strings. This ensures links with different string forms (e.g. `/creatives/1` vs `/creatives/001`) are properly deduplicated.

While `/creatives/001`-style URLs are practically impossible in our app (Rails generates integer IDs without leading zeros), the fix is trivial and eliminates the edge case entirely.

Tests: 1656 runs, 0 failures.
